app/renderer/components/window.js

```diff
@@ -27,49 +19,11 @@ const {getCurrentWindowId} = require('../currentWindow')
 window.appActions = appActions
 
 class Window extends React.Component {
-  constructor (props) {
-    super(props)
-    // initialize appState from props
-    // and then listen for updates
-    this.appState = appStoreRenderer.state
-    this.windowState = Immutable.fromJS(this.props.initWindowState) || windowStore.getState()
-    this.state = {
-      immutableData: {
-        windowState: this.windowState,
-        appState: this.appState
-      }
-    }
-    if (this.props.initWindowState) {
-      windowActions.setState(this.windowState)
-    }
-
-    this.onChange = this.onChange.bind(this)
-    this.onAppStateChange = this.onAppStateChange.bind(this)
-    windowStore.addChangeListener(this.onChange)
-    appStoreRenderer.addChangeListener(this.onAppStateChange)
-  }
-
-  componentWillMount () {
-    const activeFrameKey = this.state.immutableData.windowState.get('activeFrameKey')
-    this.props.frames.forEach((frame, i) => {
-      if (frame.guestInstanceId) {
-        appActions.newWebContentsAdded(getCurrentWindowId(), frame)
-      } else {
-        appActions.createTabRequested({
-          url: frame.location || frame.src || frame.provisionalLocation,
-          partitionNumber: frame.partitionNumber,
-          isPrivate: frame.isPrivate,
-          active: activeFrameKey ? frame.key === activeFrameKey : true,
-          discarded: frame.unloaded,
-          title: frame.title,
-          faviconUrl: frame.icon,
-          index: i
-        }, false, true /* isRestore */)
-      }
-    })
+  componentDidMount () {
+    appActions.windowReady(getCurrentWindowId())
```

Comment:
this can happen anytime after `windowStore.state = newState` and should probably go after `generateTabs`